GitUI/HelperDialogs/FormCommitDiff.cs

```diff
@@ -39,24 +40,34 @@ public FormCommitDiff(GitUICommands aCommands, string revisionGuid)
             }
         }
 
-        private void DiffFiles_SelectedIndexChanged(object sender, EventArgs e)
+        private async void DiffFiles_SelectedIndexChanged(object sender, EventArgs e)
         {
-            ViewSelectedDiff();
+            try
+            {
+                await ViewSelectedDiff();
+            }
+            catch (OperationCanceledException)
+            {
+            }
         }
 
-        private void ViewSelectedDiff()
+        private async Task ViewSelectedDiff()
         {
             if (DiffFiles.SelectedItem != null && DiffFiles.Revision != null)
             {
-                Cursor.Current = Cursors.WaitCursor;
-                DiffText.ViewChanges(DiffFiles.SelectedItemParent?.Guid, DiffFiles.Revision?.Guid, DiffFiles.SelectedItem, String.Empty);
-                Cursor.Current = Cursors.Default;
+                await DiffText.ViewChanges(DiffFiles.SelectedItemParent?.Guid, DiffFiles.Revision?.Guid, DiffFiles.SelectedItem, String.Empty);
             }
         }
 
-        void DiffText_ExtraDiffArgumentsChanged(object sender, EventArgs e)
+        async void DiffText_ExtraDiffArgumentsChanged(object sender, EventArgs e)
```

Comment:
I think this will set it off the StyleCop - please add `private` and cause unnecessary pain to @drewnoakes in #4587

Reply:
It'll conflict either way but will be easy to fix so don't hold back on my account. Thanks though.

Reply:
I rebased and applied changes